src/AppBundle/Repository/CommitteeRepository.php

```diff
@@ -227,4 +228,33 @@ private function createSitemapQb(): QueryBuilder
             ->where('c.status = :status')
             ->setParameter('status', Committee::APPROVED);
     }
+
+    /**
+     * @return Committee[]
+     */
+    public function searchCommittees(SearchParametersFilter $search): array
+    {
+        if ($coordinates = $search->getCityCoordinates()) {
+            $qb = $this
+                ->createNearbyQueryBuilder($coordinates)
+                ->andWhere($this->getNearbyExpression().' < :distance_max')
+                ->setParameter('distance_max', $search->getRadius());
+        } else {
+            $qb = $this->createQueryBuilder('c');
+        }
+
+        if (!empty($query = $search->getQuery())) {
+            $qb->andWhere('n.name like :query');
```

Comment:
Yes me too! Again I just moved this code to the repository because it was outside of it. But I agree it would be better to create the builder and pass it as an argument of repository private methods.